ts-client/src/dlmm/index.ts

```diff
@@ -3734,6 +3734,145 @@ export class DLMM {
     );
   }
 
+  public async canSyncWithMarketPrice(marketPrice: number) {
+    const marketPriceBinId = this.getBinIdFromPrice(
+      Number(
+        DLMM.getPricePerLamport(
+          this.tokenX.decimal,
+          this.tokenY.decimal,
+          marketPrice
+        )
+      ),
+      false
+    );
+    const activeBin = await this.getActiveBin();
+    const activeBinId = activeBin.binId;
+
+    const marketPriceBinArrayIndex = binIdToBinArrayIndex(
+      new BN(marketPriceBinId)
+    );
+
+    const swapForY = marketPriceBinId < activeBinId;
+    const toBinArrayIndex = findNextBinArrayIndexWithLiquidity(
+      swapForY,
+      new BN(activeBinId),
+      this.lbPair,
+      this.binArrayBitmapExtension?.account ?? null
+    );
+    if (toBinArrayIndex === null) return true;
+
+    return swapForY
+      ? marketPriceBinArrayIndex.gt(toBinArrayIndex)
+      : marketPriceBinArrayIndex.lt(toBinArrayIndex);
+  }
+
+  /**
+   * The `syncWithMarketPrice` function is used to sync the liquidity pool with the market price.
+   * @param
+   *    - `marketPrice`: The market price to sync with.
+   *    - `owner`: The public key of the owner of the liquidity pool.
+   * @returns {Promise<Transaction>}
+   */
+  public async syncWithMarketPrice(marketPrice: number, owner: PublicKey) {
+    const marketPriceBinId = this.getBinIdFromPrice(
+      Number(
+        DLMM.getPricePerLamport(
+          this.tokenX.decimal,
+          this.tokenY.decimal,
+          marketPrice
+        )
+      ),
+      false
+    );
+    const activeBin = await this.getActiveBin();
+    const activeBinId = activeBin.binId;
+
+    const marketPriceBinArrayIndex = binIdToBinArrayIndex(
+      new BN(marketPriceBinId)
+    );
+    const fromBinArrayIndex = binIdToBinArrayIndex(new BN(activeBinId));
+
+    const swapForY = marketPriceBinId < activeBinId;
+    const toBinArrayIndex = findNextBinArrayIndexWithLiquidity(
+      swapForY,
+      new BN(activeBinId),
+      this.lbPair,
+      this.binArrayBitmapExtension?.account ?? null
+    );
+    if (
+      toBinArrayIndex !== null &&
+      (swapForY
+        ? marketPriceBinArrayIndex.lt(toBinArrayIndex)
+        : marketPriceBinArrayIndex.gt(toBinArrayIndex))
+    ) {
+      throw new Error(
```

Comment:
Why not use the function canSyncWithMarketPrice? so you can keep logic in this function clean (just handle logic and no checking)